starter-code/main.js

```diff
@@ -53,6 +54,203 @@ mongoClient.connect(`mongodb://localhost:27017/crunchbase`, (error, db) => {
+              }
+            })
+            break;
+
+          case "9":
+            // 9.- How many employees has Facebook?
+            db.collection('companies').find({ name: "Facebook" }, { number_of_employees: 1 }).toArray((error, result) => {
+              if (error) {
+                console.log(error);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              } else {
+                console.log(result);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              }
+            })
+            break;
+
+          case "10":
+            // 10.- List the name of all the products of Facebook
+            db.collection('companies').find({ name: "Facebook" }, { "products.name": 1 }).toArray((error, result) => {
+              if (error) {
+                console.log(error);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              } else {
+                result.forEach(elm => {
+                  console.log(elm);
+                })
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              }
+            })
+            break;
+
+          case "11":
+            // 11.- List the people that are working at Facebook right now (check relationships field)
+            db.collection('companies').find({ name: "Facebook" }, { relationships: 1 }).toArray((error, result) => {
+              if (error) {
+                console.log(error);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              } else {
+                result.forEach(elm => {
+                  if (!elm.is_past) {
+                    console.log(elm);
+                  }
+                })
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              }
+            })
+            break;
+
+          case "12":
+            // 12.- List all the companies where "david-ebersman" has worked.
+            db.collection('companies').find({ "relationships.person.permalink": "david-ebersman" }, { name: 1 }).toArray((error, result) => {
+              if (error) {
+                console.log(error);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              } else {
+                console.log(result);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              }
+            })
+            break;
+
+          case "13":
+            // 13.- List by name the competitors of Facebook
+            db.collection('companies').find({ name: "Facebook" }, { "competitions.competitor.name": 1 }).toArray((error, result) => {
+              if (error) {
+                console.log(error);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              } else {
+                result.forEach(elm => {
```

Comment:
Mongo tiene su propio método .map que se puede emplear para esto.

```
	db.collection('companies')
		.find({ name: 'Facebook' }, { name: 1, _id: 0, competitions: 1 })
		.map(doc => doc.competitions.map(elm => elm.competitor.name))
```

Esto es un poco too much, es solo para que lo veáis.